rust/protocol/src/curve.rs

```diff
@@ -102,10 +102,7 @@ impl PublicKey {
         match self.key {
             PublicKeyData::DjbPublicKey(pub_key) => {
                 if signature.len() != 64 {
-                    return Err(SignalProtocolError::MismatchedSignatureLengthForKey(
-                        KeyType::Djb,
-                        signature.len(),
-                    ));
+                    return Ok(false);
```

Comment:
This seems like a bad change to me. If clients want to handle invalid signatures the same way as valid-but-incorrect signatures, that should be up to them, but the actual implementation in libsignal-protocol should keep those separate.

Reply:
Neither libsignal-protocol-java nor SignalProtocolKit make special exception for the wrong length, they just return false. The C library used by Android has an explicit check on the length and jumps immediately to reject. SignalProtocolKit, AFAICT, just assumes the signature is at least 64 bytes and uses whatever random data follows as the signature, which will also cause it to fail to verify.

Also there are a lot of forms of invalid signature (for example r/s can be greater than the group order) I'm not sure why the length should be considered special here.

Reply:
All right, withdrawn.